### ISSUE TITLE
RStudio crashing when calling python file through Reticulate

### ISSUE THREAD

**Issue by bormanjo** [rstudio/rstudio#3249]
### System details

    RStudio Edition : Server
    RStudio Version : 1.1.453
    OS Version      : Debian GNU/Linux 9 (Docker)
    R Version       : 3.5.1

### Steps to reproduce the problem

I have a short python file titled `connect.py` containing the following code:

``` python
import psycopg2
import pandas as pd

try:
  connect_str = "dbname='<ins_name>' user='<ins_usr>' host='<ins_hst>' password='<ins_pswd>'"
  conn = psycopg2.connect(connect_str)
  cursor = conn.cursor()
  cursor.execute("""SELECT * from <ins_tblname> LIMIT 100""")
  df = pd.DataFrame(cursor.fetchall())
  print(df.head())
except Exception as e:
  print("Uh oh, can't connect. Invalid dbname, user or password?")
  print(e)
```

When I run the following via  **`reticulate` in RStudio**:

``` R
library(reticulate)
use_python("/usr/local/bin/python")
py_config()
py_run_file("connect.py")
```

RStudio crashes

<img width="904" alt="screen shot 2018-08-01 at 6 16 46 pm" src="https://user-images.githubusercontent.com/12824247/43552038-1d1f0316-95b7-11e8-9319-18b8942446f0.png">

However, I can run this from the terminal fine when I call the R-Console manually via `$R`:

<img width="905" alt="screen shot 2018-08-01 at 6 19 51 pm" src="https://user-images.githubusercontent.com/12824247/43552215-b3276bd2-95b7-11e8-8eae-0583c3d925f5.png">

Again, when this is run directly in the console of Jupyter Lab I'm able to access the output of `py$df`

<img width="566" alt="screen shot 2018-08-01 at 6 30 27 pm" src="https://user-images.githubusercontent.com/12824247/43552607-220ff6d0-95b9-11e8-86a8-69e250a82622.png">

### Describe the problem in detail

To me, this is quite an obscure bug that I feel has to do with something happening in the background of RStudio's interaction with R. I'm really lost as to what specifically is causing this. There are 3 runtime warnings (2 from numpy and 1 from psycopg2) but these should be non-issue according to the warnings themselves and a brief google search.

### Describe the behavior you expected

I would expect that after running the code located in `connect.py` that a data.frame -like object would be available/accessible in `py$df` as shown in the last image from the Jupyter Lab environment. Again, this leads me to believe that the root cause is a bug within RStudio.

1. The output of sessionInfo() 

```
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 9 (stretch)

Matrix products: default
BLAS: /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C              LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reticulate_1.9

loaded via a namespace (and not attached):
[1] compiler_3.5.1  Matrix_1.2-14   tools_3.5.1     yaml_2.1.19     Rcpp_0.12.18    grid_3.5.1      jsonlite_1.5   
[8] lattice_0.20-35
```

2. The R code in question

``` R
library(reticulate)
use_python("/usr/local/bin/python")
py_config()
py_run_file("connect.py")
```

One last note: I am able to run python code in the first place using the `py_run_file` command for a python file containing the following:

``` python
import numpy
print(numpy.array([1,2,3]))
a = numpy.array([1,2,3])
```

Where `a` becomes available in `py$a`.

**Comment by ronblum** [comment]
@bormanjo Thanks for reporting this. I am getting something somewhat different when running in a similar configuration (python3, R 3.5.0), with similar code connecting to a PostgreSQL database. I am still testing it. However, it runs fine with R 3.4.4. Would you be willing to try out that version of R and see what happens? Thanks!

**Comment by jmcphers** [comment]
Haven't heard back in over a year; closing due to inactivity.

**Comment by reyar** [comment]
Hi RStudio team,

I have a problem which might be correlated to the problem described by @bormanjo:

System details:

```
> sessionInfo()
R version 3.6.2 (2019-12-12)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 16299)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_1.4.0     reticulate_1.14

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3      digest_0.6.23   later_1.0.0     mime_0.8        rappdirs_0.3.1  R6_2.4.1        xtable_1.8-4   
 [8] jsonlite_1.6    magrittr_1.5    rlang_0.4.2     promises_1.1.0  tools_3.6.2     httpuv_1.5.2    fastmap_1.0.1  
[15] compiler_3.6.2  htmltools_0.4.0
```

**The problem:**
I have a a shiny app calling a python file through py_run_file. When I launch the app the first time in a  new R session, it works as expected. After stopping the app and launching it again the second time, it crashes (stops without delivered errors). The third time, it works. The fourth time, it crashes, etc. It does not depend on the content of the python file. Moreover, I can call the python file without any problems from the R console in RStudio. Morover, if I call the python file from the R console, after a successful launch of the app (and before a potential non-successful launch), then the problem dissapears (the next launch will be also successful).  

**reprex**

test.py
```
import numpy as np

a=5
b=np.array([1,2,3])
```

server.R
```
library(shiny)
library(reticulate)

shinyServer(function(input, output, session) {
    
  observe({
       py_run_file("test.py")
       output$a<- renderText({paste(py$a)})
     
   })
  
})
```

ui.R
```
library(shiny)

shinyUI(fluidPage(
  
  fluidRow( 
    
    column(12, 
           
      h3("Test"),
      h3(textOutput("a"))
           
           
    )
    
  )
  
))
```

**Further details**
The problem is new, it arrised after an update of the R libraries on the 2. January 2020. The previous update was on 3. December 2019. I have removed and installed the older versions of packages which were updated in the meantime (for example reticulate) - it did not help. I cannot find the cause of the problem. 

Please help!

Thanks!

**Comment by ronblum** [comment]
@reyar Thank you for letting us know and for providing the session info! I'll reopen this issue. We might need to open a new issue instead, but I'll do this for now.

What version of RStudio are you using? Also, when it crashes, is it with the same "R Session Error" window?

    RStudio Edition : Desktop
    RStudio Version : 
    OS Version      : Windows 10
    R Version       : 3.6.2

Note: not reproducible on RStudio Desktop 1.2.5033 on MacOS 10.15 with R 3.6.1.

**Comment by reyar** [comment]
Hi @ronblum 

Thanks for the reply! 

I am using RStudio Desktop 1.2.5033. When it crashes, the debug location points at `py_run_file_impl`. No, I did not have any issues with Python outside the R environment and my other Shiny apps which do not call Python work as expected. Currently, my only practical solution is to restart R within RStudio before each runApp().  

I opened an issue with this problem here: https://github.com/rstudio/reticulate/issues/683
The system described there is slightly different but very similar. The installation of the `reticulate` from github solved the problem for RGUi but unfortunately not for RStudio.
